src/main/java/bookstore/controller/OrderController.java

```diff
@@ -0,0 +1,116 @@
+import bookstore.service.OrderService;
+import bookstore.util.SortParametersParsingUtil;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.ExampleObject;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/orders")
+@RequiredArgsConstructor
+@ControllerAdvice
+@Tag(name = "Order Controller", description = "Endpoints for managing orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final OrderItemService orderItemService;
+    private final SortParametersParsingUtil sortParametersParsingUtil;
+    
+    @Operation(summary = "Place a new order")
+    @ResponseStatus(HttpStatus.CREATED)
+    @ApiResponse(responseCode = "201", description = "Order created", content = {
+            @Content(mediaType = "application/json", schema =
+            @Schema(implementation = OrderResponseDto.class),
+            examples = @ExampleObject(value = "{\n"
+                                              + "    \"id\": 1,\n"
+                                              + "    \"shippingAddress\": \"address\",\n"
+                                              + "    \"orderItems\": [\n"
+                                              + "      {\n"
+                                              + "        \"bookId\": 1,\n"
+                                              + "        \"quantity\": 2\n"
+                                              + "      }\n"
+                                              + "    ]\n"
+                                              + "}"))
+    })
+    @PostMapping
+    public OrderResponseDto placeOrder(@Valid @RequestBody OrderRequestDto order,
```

Comment:
Only the user should be able to make an order. (Any anonymous, or admins)
```suggestion
    @PostMapping
    @PreAuthorize("hasAuthority('USER')")
    public OrderResponseDto placeOrder(@Valid @RequestBody OrderRequestDto order,
```